### PR TITLE
refactor: 开发/构建流程优化

### PR DESCRIPTION
**Pull request by sansi-lcj**

### CODE REVIEW

File: plugins/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+          build: {
+            outDir: 'dist',
+            lib: {
+              entry: 'src/index.ts',
+              name: 'dnalogel',
+              formats: ['es', 'umd', 'cjs'],
+              fileName: (format, entryName) => `${entryName}.${format}.js`,
+            },
+          },
+          plugins: [cp({ targets: [{ src: 'dist/index.es.js', dest: 'dist', rename: 'index.js' }] })],
```
Comment (Emiyaaaaa) [inline plugins/vite.config.js:92]: 之前好像是把index.cjs.js改的index.js，我那版改错了

File: plugins/vite.config.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            rollupOptions: {
+              output: {
+                preserveModules: true,
+                preserveModulesRoot: 'src',
+              },
+            },
+            lib: {
+              entry: 'src/index.ts',
+              name: 'dnalogel',
+              formats: ['es'],
```
Comment (Emiyaaaaa) [inline plugins/vite.config.js:124]: 之前也是打的cjs包

Comment (Emiyaaaaa) [reply plugins/vite.config.js:124]: https://github.com/realsee-developer/dnalogel/blob/9031a316477296f290efd9d34a7986b3885669a1/plugins/rollup.config.js

File: plugins/vite.config.js (hunk 1 above)
Comment (Emiyaaaaa) [reply plugins/vite.config.js:92]: https://github.com/realsee-developer/dnalogel/blob/9031a316477296f290efd9d34a7986b3885669a1/plugins/rollup.config.js

**Review by Emiyaaaaa — CHANGES_REQUESTED** [review]

**Review by Arrrrray — APPROVED** [review]

File: plugins/vite.config.js (hunk 1 above)
Comment (sansi-lcj) [reply plugins/vite.config.js]: 已处理

Comment (sansi-lcj) [reply plugins/vite.config.js]: 已处理

**Review by Emiyaaaaa — APPROVED** [review]
LGTM

**Pull request merged**